### PR TITLE
feat: support networking resources in export-terraform

### PR DESCRIPTION
**Pull request by bharathkkb**

### CODE REVIEW

**Review by mengqiy — APPROVED** [review]

**Comment by mengqiy** [comment]
@bharathkkb Should we ask someone from your team to take another look at the logic?

File: functions/go/export-terraform/terraformgenerator/resources.go [hunk 1]
```diff
@@ -148,8 +175,23 @@ func (resource *terraformResource) GetOrganization() *terraformResource {
...
 type referencePath struct {
-	kind string // If kind is unset, it will be auto-detected
-	path []string
+	kind            string // If kind is unset, it will be auto-detected
+	path            []string
+	customRetriever func(*sdk.KubeObject) string // custom func to retrieve ref from KubeObject
+}
+
+// singleCACustomRetriever creates a customRetriever for a compute address ref
+func singleCACustomRetriever(path []string) func(*sdk.KubeObject) string {
```
Comment (g-awmalik) [inline functions/go/export-terraform/terraformgenerator/resources.go:184]: My first impression of CA in the function name was "certificate authority" which this is not related to. Can this be called `getCustomComputeAddress` or something similar?

**Review by g-awmalik — APPROVED** [review]
LGTM. Few nits.

File: functions/go/export-terraform/terraformgenerator/resources.go (hunk 1 above)
Comment (bharathkkb) [reply functions/go/export-terraform/terraformgenerator/resources.go]: Updated to use `singleComputeAddressRetriever` instead

File: functions/go/export-terraform/terraformgenerator/network.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	sdk "github.com/GoogleContainerTools/kpt-functions-catalog/thirdparty/kyaml/fnsdk"
+)
+
+type firewallAllow struct {
+	Ports    []string `yaml:"ports,omitempty"`
+	Protocol string   `yaml:"protocol"`
+}
+
+func (resource *terraformResource) GetFirewallAllow() []firewallAllow {
```
[the remark this answers is not in the document]
Comment (bharathkkb) [reply functions/go/export-terraform/terraformgenerator/network.go]: Updated to `GetFirewallAllowPortsProtocol` since it has both

**Review by g-awmalik — APPROVED** [review]
LGTM

**Comment by bharathkkb** [comment]
@mengqiy looks like I don't have permissions to merge with approvals, PTAL when you get a chance.
